apps/checkout-app/backend/payments/updateOrCreateTransaction.ts

```diff
@@ -0,0 +1,47 @@
+import { TransactionCreateMutationVariables } from "@/checkout-app/graphql";
+import { createTransaction } from "./createTransaction";
+import { getOrderTransactions } from "./getOrderTransactions";
+import { updateTransaction } from "./updateTransaction";
+
+async function findTransactionInOrder(
+  reference: string | null | undefined,
+  orderId: string
+) {
+  if (!reference) return null;
+
+  try {
+    const orderTransactions = await getOrderTransactions({ id: orderId });
+
+    if (orderTransactions.length) {
+      return orderTransactions.find(
+        (transaction) => transaction.reference === reference
+      );
+    }
+  } catch (e) {
+    console.error("Error while finding transaction in order", e);
```

Comment:
Error can mean many different things could happen, maybe simply `Could not find transaction in order`?

Reply:
This could be interpreted as "we didn't find a transaction in order", which isn't true because in that case we simply return `undefined` from that function. What actually happens is: "something bad happened when making request to Saleor to fetch transactions in order". This unspecific "error" because we really don't know what happened 😅